### ISSUE TITLE
Create Datavase to store results

### ISSUE THREAD

**Issue by marshalmiller** [rottingresearch/rottingresearch#125]
I think it would be extremely helpful if we were to store the results from each analysis in a database. I do not want to store user data. But things like the date written (pulled from metadata), and data submitted to us. Overall links, DOI links, Error Codes, and such. This would allow us to keep long-term stats for further research. Even how the same documents change over time. We could then create a dashboard to share those results.

**Comment by marshalmiller** [comment]
To be clear, we don't need assistance running the database, but making the connections and correct commands to make the database do what we want it to.

**Comment by aditirao7** [comment]
Maybe we can consider MongoDB for this?

**Comment by marshalmiller** [comment]
I have no preference for SQL vs. NoSQL, as long as it addresses the need.

**Comment by aditirao7** [comment]
Since we're dealing with a lot of textual data and the long term goal is data analytics, MongoDB would be useful. I believe it provides dashboards and APIs for analytics

**Comment by aditirao7** [comment]
@marshalmiller How would you like the data to be stored? Do you have a high level schema in mind? I was thinking we could have a model "Analysis" where the required data associated with every analysis can be stored. Also, in order to compare documents we would need some unique attribute associated with the document, i.e how do we know that 2 different analyses are linked to the same document?

**Comment by marshalmiller** [comment]
@aditirao7 I think it would be a good idea to append the timestamp to the filename. This way, we "could" also compare the same document at different points in time. Like if it was just a year apart, did the results change? I assume that would be unique every time. These are just initial thoughts.

"title" (pulled from PDF)
"author" (pulled from PDF)
"date-created" (pulled from PDF)
"dois" (number)
"dois-broken" (number)
"URLs" (I'd love to collect all of the links found. This way we can find sites that commonly break)
"links" (number)
"links-broken" (number)
"title+date-checked"
"arxiv" (number)
"arxiv-broken" (number)
"code-returned" (Which Code was returned. Insert NULL if not one of the regular ones. Like this could be where we weed out those ones that have issues) (number)

**Comment by aditirao7** [comment]
@marshalmiller Yep that would be unique but if the user changed the filename it wouldn't work right? Maybe creation date and author can be used to distinguish? The rest looks good to me.

**Comment by marshalmiller** [comment]
I would prefer that we keep the time stamp in some way. I'm okay with it being in combo with anything. But I can see the timestamp of the upload as being especially important in the future. I'm thinking of more future updates and such. For instance, The Internet Archive uses timestamps for their Wayback machine. So we could pull info like that. But I'm okay with the Unique Identifier being anything in addition to the time stamp. I don't actually want to stop people from uploading the exact same file every time. I'd like that, so we can detect changes in the links over time. Does that give you any ideas? I'm mostly open to anything; I'd just like the timestamp to be included in the unique identifier.

**Comment by aditirao7** [comment]
Yep, I was thinking about using MongoDB which inherently records timestamps for every instance. So we wouldn't have to append it separately but we can still extract it when necessary.
